# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libGLESv2/entry_points_gles_3_1_autogen.cpp
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libGLESv2/entry_points_gles_3_1_autogen.cpp
--- a/gfx/angle/checkout/src/libGLESv2/entry_points_gles_3_1_autogen.cpp	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libGLESv2/entry_points_gles_3_1_autogen.cpp	Sat Sep 07 13:38:36 2019 +0000
@@ -12,20 +12,21 @@
 
 #include "libANGLE/Context.h"
 #include "libANGLE/Context.inl.h"
+#include "libANGLE/capture_gles_3_1_autogen.h"
+#include "libANGLE/entry_points_utils.h"
 #include "libANGLE/validationES31.h"
-#include "libGLESv2/entry_points_utils.h"
 #include "libGLESv2/global_state.h"
 
 namespace gl
 {
 void GL_APIENTRY ActiveShaderProgram(GLuint pipeline, GLuint program)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint pipeline = %u, GLuint program = %u)", pipeline, program);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ActiveShaderProgram, context, pipeline, program);
         if (context->skipValidation() || ValidateActiveShaderProgram(context, pipeline, program))
         {
             context->activeShaderProgram(pipeline, program);
@@ -41,7 +42,6 @@
                                   GLenum access,
                                   GLenum format)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint unit = %u, GLuint texture = %u, GLint level = %d, GLboolean layered = %u, GLint "
         "layer = %d, GLenum access = 0x%X, GLenum format = 0x%X)",
@@ -50,6 +50,8 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(BindImageTexture, context, unit, texture, level, layered, layer, access,
+                      format);
         if (context->skipValidation() ||
             ValidateBindImageTexture(context, unit, texture, level, layered, layer, access, format))
         {
@@ -60,12 +62,12 @@
 
 void GL_APIENTRY BindProgramPipeline(GLuint pipeline)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint pipeline = %u)", pipeline);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(BindProgramPipeline, context, pipeline);
         if (context->skipValidation() || ValidateBindProgramPipeline(context, pipeline))
         {
             context->bindProgramPipeline(pipeline);
@@ -78,7 +80,6 @@
                                   GLintptr offset,
                                   GLsizei stride)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint bindingindex = %u, GLuint buffer = %u, GLintptr offset = %llu, GLsizei stride = "
         "%d)",
@@ -87,6 +88,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(BindVertexBuffer, context, bindingindex, buffer, offset, stride);
         if (context->skipValidation() ||
             ValidateBindVertexBuffer(context, bindingindex, buffer, offset, stride))
         {
@@ -97,7 +99,6 @@
 
 GLuint GL_APIENTRY CreateShaderProgramv(GLenum type, GLsizei count, const GLchar *const *strings)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum type = 0x%X, GLsizei count = %d, const GLchar *const*strings = 0x%016" PRIxPTR
           ")",
           type, count, (uintptr_t)strings);
@@ -106,6 +107,7 @@
     if (context)
     {
         ShaderType typePacked = FromGLenum<ShaderType>(type);
+        ANGLE_CAPTURE(CreateShaderProgramv, context, typePacked, count, strings);
         if (context->skipValidation() ||
             ValidateCreateShaderProgramv(context, typePacked, count, strings))
         {
@@ -118,12 +120,12 @@
 
 void GL_APIENTRY DeleteProgramPipelines(GLsizei n, const GLuint *pipelines)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLsizei n = %d, const GLuint *pipelines = 0x%016" PRIxPTR ")", n, (uintptr_t)pipelines);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(DeleteProgramPipelines, context, n, pipelines);
         if (context->skipValidation() || ValidateDeleteProgramPipelines(context, n, pipelines))
         {
             context->deleteProgramPipelines(n, pipelines);
@@ -133,13 +135,13 @@
 
 void GL_APIENTRY DispatchCompute(GLuint num_groups_x, GLuint num_groups_y, GLuint num_groups_z)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint num_groups_x = %u, GLuint num_groups_y = %u, GLuint num_groups_z = %u)",
           num_groups_x, num_groups_y, num_groups_z);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(DispatchCompute, context, num_groups_x, num_groups_y, num_groups_z);
         if (context->skipValidation() ||
             ValidateDispatchCompute(context, num_groups_x, num_groups_y, num_groups_z))
         {
@@ -150,12 +152,12 @@
 
 void GL_APIENTRY DispatchComputeIndirect(GLintptr indirect)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLintptr indirect = %llu)", static_cast<unsigned long long>(indirect));
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(DispatchComputeIndirect, context, indirect);
         if (context->skipValidation() || ValidateDispatchComputeIndirect(context, indirect))
         {
             context->dispatchComputeIndirect(indirect);
@@ -165,7 +167,6 @@
 
 void GL_APIENTRY DrawArraysIndirect(GLenum mode, const void *indirect)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum mode = 0x%X, const void *indirect = 0x%016" PRIxPTR ")", mode,
           (uintptr_t)indirect);
 
@@ -173,6 +174,7 @@
     if (context)
     {
         PrimitiveMode modePacked = FromGLenum<PrimitiveMode>(mode);
+        ANGLE_CAPTURE(DrawArraysIndirect, context, modePacked, indirect);
         if (context->skipValidation() || ValidateDrawArraysIndirect(context, modePacked, indirect))
         {
             context->drawArraysIndirect(modePacked, indirect);
@@ -182,7 +184,6 @@
 
 void GL_APIENTRY DrawElementsIndirect(GLenum mode, GLenum type, const void *indirect)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum mode = 0x%X, GLenum type = 0x%X, const void *indirect = 0x%016" PRIxPTR ")",
           mode, type, (uintptr_t)indirect);
 
@@ -191,6 +192,7 @@
     {
         PrimitiveMode modePacked    = FromGLenum<PrimitiveMode>(mode);
         DrawElementsType typePacked = FromGLenum<DrawElementsType>(type);
+        ANGLE_CAPTURE(DrawElementsIndirect, context, modePacked, typePacked, indirect);
         if (context->skipValidation() ||
             ValidateDrawElementsIndirect(context, modePacked, typePacked, indirect))
         {
@@ -201,12 +203,12 @@
 
 void GL_APIENTRY FramebufferParameteri(GLenum target, GLenum pname, GLint param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLint param = %d)", target, pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(FramebufferParameteri, context, target, pname, param);
         if (context->skipValidation() ||
             ValidateFramebufferParameteri(context, target, pname, param))
         {
@@ -217,12 +219,12 @@
 
 void GL_APIENTRY GenProgramPipelines(GLsizei n, GLuint *pipelines)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLsizei n = %d, GLuint *pipelines = 0x%016" PRIxPTR ")", n, (uintptr_t)pipelines);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GenProgramPipelines, context, n, pipelines);
         if (context->skipValidation() || ValidateGenProgramPipelines(context, n, pipelines))
         {
             context->genProgramPipelines(n, pipelines);
@@ -232,13 +234,13 @@
 
 void GL_APIENTRY GetBooleani_v(GLenum target, GLuint index, GLboolean *data)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLuint index = %u, GLboolean *data = 0x%016" PRIxPTR ")", target,
           index, (uintptr_t)data);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetBooleani_v, context, target, index, data);
         if (context->skipValidation() || ValidateGetBooleani_v(context, target, index, data))
         {
             context->getBooleani_v(target, index, data);
@@ -248,13 +250,13 @@
 
 void GL_APIENTRY GetFramebufferParameteriv(GLenum target, GLenum pname, GLint *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLint *params = 0x%016" PRIxPTR ")", target,
           pname, (uintptr_t)params);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetFramebufferParameteriv, context, target, pname, params);
         if (context->skipValidation() ||
             ValidateGetFramebufferParameteriv(context, target, pname, params))
         {
@@ -265,13 +267,13 @@
 
 void GL_APIENTRY GetMultisamplefv(GLenum pname, GLuint index, GLfloat *val)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, GLuint index = %u, GLfloat *val = 0x%016" PRIxPTR ")", pname,
           index, (uintptr_t)val);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetMultisamplefv, context, pname, index, val);
         if (context->skipValidation() || ValidateGetMultisamplefv(context, pname, index, val))
         {
             context->getMultisamplefv(pname, index, val);
@@ -284,7 +286,6 @@
                                        GLenum pname,
                                        GLint *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLenum programInterface = 0x%X, GLenum pname = 0x%X, GLint *params "
         "= 0x%016" PRIxPTR ")",
@@ -293,6 +294,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetProgramInterfaceiv, context, program, programInterface, pname, params);
         if (context->skipValidation() ||
             ValidateGetProgramInterfaceiv(context, program, programInterface, pname, params))
         {
@@ -306,7 +308,6 @@
                                            GLsizei *length,
                                            GLchar *infoLog)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint pipeline = %u, GLsizei bufSize = %d, GLsizei *length = 0x%016" PRIxPTR
           ", GLchar *infoLog = 0x%016" PRIxPTR ")",
           pipeline, bufSize, (uintptr_t)length, (uintptr_t)infoLog);
@@ -314,6 +315,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetProgramPipelineInfoLog, context, pipeline, bufSize, length, infoLog);
         if (context->skipValidation() ||
             ValidateGetProgramPipelineInfoLog(context, pipeline, bufSize, length, infoLog))
         {
@@ -324,13 +326,13 @@
 
 void GL_APIENTRY GetProgramPipelineiv(GLuint pipeline, GLenum pname, GLint *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint pipeline = %u, GLenum pname = 0x%X, GLint *params = 0x%016" PRIxPTR ")",
           pipeline, pname, (uintptr_t)params);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetProgramPipelineiv, context, pipeline, pname, params);
         if (context->skipValidation() ||
             ValidateGetProgramPipelineiv(context, pipeline, pname, params))
         {
@@ -343,7 +345,6 @@
                                            GLenum programInterface,
                                            const GLchar *name)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLenum programInterface = 0x%X, const GLchar *name = 0x%016" PRIxPTR
         ")",
@@ -352,6 +353,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetProgramResourceIndex, context, program, programInterface, name);
         if (context->skipValidation() ||
             ValidateGetProgramResourceIndex(context, program, programInterface, name))
         {
@@ -366,7 +368,6 @@
                                              GLenum programInterface,
                                              const GLchar *name)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLenum programInterface = 0x%X, const GLchar *name = 0x%016" PRIxPTR
         ")",
@@ -375,6 +376,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetProgramResourceLocation, context, program, programInterface, name);
         if (context->skipValidation() ||
             ValidateGetProgramResourceLocation(context, program, programInterface, name))
         {
@@ -392,7 +394,6 @@
                                         GLsizei *length,
                                         GLchar *name)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLenum programInterface = 0x%X, GLuint index = %u, GLsizei bufSize "
         "= %d, GLsizei *length = 0x%016" PRIxPTR ", GLchar *name = 0x%016" PRIxPTR ")",
@@ -401,6 +402,8 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetProgramResourceName, context, program, programInterface, index, bufSize,
+                      length, name);
         if (context->skipValidation() ||
             ValidateGetProgramResourceName(context, program, programInterface, index, bufSize,
                                            length, name))
@@ -420,7 +423,6 @@
                                       GLsizei *length,
                                       GLint *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLenum programInterface = 0x%X, GLuint index = %u, GLsizei "
         "propCount = %d, const GLenum *props = 0x%016" PRIxPTR
@@ -432,6 +434,8 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetProgramResourceiv, context, program, programInterface, index, propCount,
+                      props, bufSize, length, params);
         if (context->skipValidation() ||
             ValidateGetProgramResourceiv(context, program, programInterface, index, propCount,
                                          props, bufSize, length, params))
@@ -444,7 +448,6 @@
 
 void GL_APIENTRY GetTexLevelParameterfv(GLenum target, GLint level, GLenum pname, GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLenum target = 0x%X, GLint level = %d, GLenum pname = 0x%X, GLfloat *params = "
         "0x%016" PRIxPTR ")",
@@ -454,6 +457,7 @@
     if (context)
     {
         TextureTarget targetPacked = FromGLenum<TextureTarget>(target);
+        ANGLE_CAPTURE(GetTexLevelParameterfv, context, targetPacked, level, pname, params);
         if (context->skipValidation() ||
             ValidateGetTexLevelParameterfv(context, targetPacked, level, pname, params))
         {
@@ -464,7 +468,6 @@
 
 void GL_APIENTRY GetTexLevelParameteriv(GLenum target, GLint level, GLenum pname, GLint *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLenum target = 0x%X, GLint level = %d, GLenum pname = 0x%X, GLint *params = "
         "0x%016" PRIxPTR ")",
@@ -474,6 +477,7 @@
     if (context)
     {
         TextureTarget targetPacked = FromGLenum<TextureTarget>(target);
+        ANGLE_CAPTURE(GetTexLevelParameteriv, context, targetPacked, level, pname, params);
         if (context->skipValidation() ||
             ValidateGetTexLevelParameteriv(context, targetPacked, level, pname, params))
         {
@@ -484,12 +488,12 @@
 
 GLboolean GL_APIENTRY IsProgramPipeline(GLuint pipeline)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint pipeline = %u)", pipeline);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(IsProgramPipeline, context, pipeline);
         if (context->skipValidation() || ValidateIsProgramPipeline(context, pipeline))
         {
             return context->isProgramPipeline(pipeline);
@@ -501,12 +505,12 @@
 
 void GL_APIENTRY MemoryBarrier(GLbitfield barriers)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLbitfield barriers = 0x%X)", barriers);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(MemoryBarrier, context, barriers);
         if (context->skipValidation() || ValidateMemoryBarrier(context, barriers))
         {
             context->memoryBarrier(barriers);
@@ -516,12 +520,12 @@
 
 void GL_APIENTRY MemoryBarrierByRegion(GLbitfield barriers)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLbitfield barriers = 0x%X)", barriers);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(MemoryBarrierByRegion, context, barriers);
         if (context->skipValidation() || ValidateMemoryBarrierByRegion(context, barriers))
         {
             context->memoryBarrierByRegion(barriers);
@@ -531,12 +535,12 @@
 
 void GL_APIENTRY ProgramUniform1f(GLuint program, GLint location, GLfloat v0)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint program = %u, GLint location = %d, GLfloat v0 = %f)", program, location, v0);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform1f, context, program, location, v0);
         if (context->skipValidation() || ValidateProgramUniform1f(context, program, location, v0))
         {
             context->programUniform1f(program, location, v0);
@@ -549,7 +553,6 @@
                                    GLsizei count,
                                    const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLfloat *value = "
         "0x%016" PRIxPTR ")",
@@ -558,6 +561,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform1fv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform1fv(context, program, location, count, value))
         {
@@ -568,12 +572,12 @@
 
 void GL_APIENTRY ProgramUniform1i(GLuint program, GLint location, GLint v0)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint program = %u, GLint location = %d, GLint v0 = %d)", program, location, v0);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform1i, context, program, location, v0);
         if (context->skipValidation() || ValidateProgramUniform1i(context, program, location, v0))
         {
             context->programUniform1i(program, location, v0);
@@ -586,7 +590,6 @@
                                    GLsizei count,
                                    const GLint *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLint *value = "
         "0x%016" PRIxPTR ")",
@@ -595,6 +598,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform1iv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform1iv(context, program, location, count, value))
         {
@@ -605,12 +609,12 @@
 
 void GL_APIENTRY ProgramUniform1ui(GLuint program, GLint location, GLuint v0)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint program = %u, GLint location = %d, GLuint v0 = %u)", program, location, v0);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform1ui, context, program, location, v0);
         if (context->skipValidation() || ValidateProgramUniform1ui(context, program, location, v0))
         {
             context->programUniform1ui(program, location, v0);
@@ -623,7 +627,6 @@
                                     GLsizei count,
                                     const GLuint *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLuint *value = "
         "0x%016" PRIxPTR ")",
@@ -632,6 +635,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform1uiv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform1uiv(context, program, location, count, value))
         {
@@ -642,13 +646,13 @@
 
 void GL_APIENTRY ProgramUniform2f(GLuint program, GLint location, GLfloat v0, GLfloat v1)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint program = %u, GLint location = %d, GLfloat v0 = %f, GLfloat v1 = %f)", program,
           location, v0, v1);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform2f, context, program, location, v0, v1);
         if (context->skipValidation() ||
             ValidateProgramUniform2f(context, program, location, v0, v1))
         {
@@ -662,7 +666,6 @@
                                    GLsizei count,
                                    const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLfloat *value = "
         "0x%016" PRIxPTR ")",
@@ -671,6 +674,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform2fv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform2fv(context, program, location, count, value))
         {
@@ -681,13 +685,13 @@
 
 void GL_APIENTRY ProgramUniform2i(GLuint program, GLint location, GLint v0, GLint v1)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint program = %u, GLint location = %d, GLint v0 = %d, GLint v1 = %d)", program,
           location, v0, v1);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform2i, context, program, location, v0, v1);
         if (context->skipValidation() ||
             ValidateProgramUniform2i(context, program, location, v0, v1))
         {
@@ -701,7 +705,6 @@
                                    GLsizei count,
                                    const GLint *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLint *value = "
         "0x%016" PRIxPTR ")",
@@ -710,6 +713,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform2iv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform2iv(context, program, location, count, value))
         {
@@ -720,13 +724,13 @@
 
 void GL_APIENTRY ProgramUniform2ui(GLuint program, GLint location, GLuint v0, GLuint v1)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint program = %u, GLint location = %d, GLuint v0 = %u, GLuint v1 = %u)", program,
           location, v0, v1);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform2ui, context, program, location, v0, v1);
         if (context->skipValidation() ||
             ValidateProgramUniform2ui(context, program, location, v0, v1))
         {
@@ -740,7 +744,6 @@
                                     GLsizei count,
                                     const GLuint *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLuint *value = "
         "0x%016" PRIxPTR ")",
@@ -749,6 +752,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform2uiv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform2uiv(context, program, location, count, value))
         {
@@ -760,7 +764,6 @@
 void GL_APIENTRY
 ProgramUniform3f(GLuint program, GLint location, GLfloat v0, GLfloat v1, GLfloat v2)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLfloat v0 = %f, GLfloat v1 = %f, GLfloat v2 = "
         "%f)",
@@ -769,6 +772,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform3f, context, program, location, v0, v1, v2);
         if (context->skipValidation() ||
             ValidateProgramUniform3f(context, program, location, v0, v1, v2))
         {
@@ -782,7 +786,6 @@
                                    GLsizei count,
                                    const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLfloat *value = "
         "0x%016" PRIxPTR ")",
@@ -791,6 +794,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform3fv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform3fv(context, program, location, count, value))
         {
@@ -801,13 +805,13 @@
 
 void GL_APIENTRY ProgramUniform3i(GLuint program, GLint location, GLint v0, GLint v1, GLint v2)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint program = %u, GLint location = %d, GLint v0 = %d, GLint v1 = %d, GLint v2 = %d)",
           program, location, v0, v1, v2);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform3i, context, program, location, v0, v1, v2);
         if (context->skipValidation() ||
             ValidateProgramUniform3i(context, program, location, v0, v1, v2))
         {
@@ -821,7 +825,6 @@
                                    GLsizei count,
                                    const GLint *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLint *value = "
         "0x%016" PRIxPTR ")",
@@ -830,6 +833,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform3iv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform3iv(context, program, location, count, value))
         {
@@ -840,7 +844,6 @@
 
 void GL_APIENTRY ProgramUniform3ui(GLuint program, GLint location, GLuint v0, GLuint v1, GLuint v2)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLuint v0 = %u, GLuint v1 = %u, GLuint v2 = "
         "%u)",
@@ -849,6 +852,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform3ui, context, program, location, v0, v1, v2);
         if (context->skipValidation() ||
             ValidateProgramUniform3ui(context, program, location, v0, v1, v2))
         {
@@ -862,7 +866,6 @@
                                     GLsizei count,
                                     const GLuint *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLuint *value = "
         "0x%016" PRIxPTR ")",
@@ -871,6 +874,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform3uiv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform3uiv(context, program, location, count, value))
         {
@@ -882,7 +886,6 @@
 void GL_APIENTRY
 ProgramUniform4f(GLuint program, GLint location, GLfloat v0, GLfloat v1, GLfloat v2, GLfloat v3)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLfloat v0 = %f, GLfloat v1 = %f, GLfloat v2 = "
         "%f, GLfloat v3 = %f)",
@@ -891,6 +894,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform4f, context, program, location, v0, v1, v2, v3);
         if (context->skipValidation() ||
             ValidateProgramUniform4f(context, program, location, v0, v1, v2, v3))
         {
@@ -904,7 +908,6 @@
                                    GLsizei count,
                                    const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLfloat *value = "
         "0x%016" PRIxPTR ")",
@@ -913,6 +916,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform4fv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform4fv(context, program, location, count, value))
         {
@@ -924,7 +928,6 @@
 void GL_APIENTRY
 ProgramUniform4i(GLuint program, GLint location, GLint v0, GLint v1, GLint v2, GLint v3)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLint v0 = %d, GLint v1 = %d, GLint v2 = %d, "
         "GLint v3 = %d)",
@@ -933,6 +936,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform4i, context, program, location, v0, v1, v2, v3);
         if (context->skipValidation() ||
             ValidateProgramUniform4i(context, program, location, v0, v1, v2, v3))
         {
@@ -946,7 +950,6 @@
                                    GLsizei count,
                                    const GLint *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLint *value = "
         "0x%016" PRIxPTR ")",
@@ -955,6 +958,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform4iv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform4iv(context, program, location, count, value))
         {
@@ -966,7 +970,6 @@
 void GL_APIENTRY
 ProgramUniform4ui(GLuint program, GLint location, GLuint v0, GLuint v1, GLuint v2, GLuint v3)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLuint v0 = %u, GLuint v1 = %u, GLuint v2 = "
         "%u, GLuint v3 = %u)",
@@ -975,6 +978,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform4ui, context, program, location, v0, v1, v2, v3);
         if (context->skipValidation() ||
             ValidateProgramUniform4ui(context, program, location, v0, v1, v2, v3))
         {
@@ -988,7 +992,6 @@
                                     GLsizei count,
                                     const GLuint *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, const GLuint *value = "
         "0x%016" PRIxPTR ")",
@@ -997,6 +1000,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniform4uiv, context, program, location, count, value);
         if (context->skipValidation() ||
             ValidateProgramUniform4uiv(context, program, location, count, value))
         {
@@ -1011,7 +1015,6 @@
                                          GLboolean transpose,
                                          const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, GLboolean transpose = %u, "
         "const GLfloat *value = 0x%016" PRIxPTR ")",
@@ -1020,6 +1023,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniformMatrix2fv, context, program, location, count, transpose, value);
         if (context->skipValidation() ||
             ValidateProgramUniformMatrix2fv(context, program, location, count, transpose, value))
         {
@@ -1034,7 +1038,6 @@
                                            GLboolean transpose,
                                            const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, GLboolean transpose = %u, "
         "const GLfloat *value = 0x%016" PRIxPTR ")",
@@ -1043,6 +1046,8 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniformMatrix2x3fv, context, program, location, count, transpose,
+                      value);
         if (context->skipValidation() ||
             ValidateProgramUniformMatrix2x3fv(context, program, location, count, transpose, value))
         {
@@ -1057,7 +1062,6 @@
                                            GLboolean transpose,
                                            const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, GLboolean transpose = %u, "
         "const GLfloat *value = 0x%016" PRIxPTR ")",
@@ -1066,6 +1070,8 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniformMatrix2x4fv, context, program, location, count, transpose,
+                      value);
         if (context->skipValidation() ||
             ValidateProgramUniformMatrix2x4fv(context, program, location, count, transpose, value))
         {
@@ -1080,7 +1086,6 @@
                                          GLboolean transpose,
                                          const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, GLboolean transpose = %u, "
         "const GLfloat *value = 0x%016" PRIxPTR ")",
@@ -1089,6 +1094,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniformMatrix3fv, context, program, location, count, transpose, value);
         if (context->skipValidation() ||
             ValidateProgramUniformMatrix3fv(context, program, location, count, transpose, value))
         {
@@ -1103,7 +1109,6 @@
                                            GLboolean transpose,
                                            const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, GLboolean transpose = %u, "
         "const GLfloat *value = 0x%016" PRIxPTR ")",
@@ -1112,6 +1117,8 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniformMatrix3x2fv, context, program, location, count, transpose,
+                      value);
         if (context->skipValidation() ||
             ValidateProgramUniformMatrix3x2fv(context, program, location, count, transpose, value))
         {
@@ -1126,7 +1133,6 @@
                                            GLboolean transpose,
                                            const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, GLboolean transpose = %u, "
         "const GLfloat *value = 0x%016" PRIxPTR ")",
@@ -1135,6 +1141,8 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniformMatrix3x4fv, context, program, location, count, transpose,
+                      value);
         if (context->skipValidation() ||
             ValidateProgramUniformMatrix3x4fv(context, program, location, count, transpose, value))
         {
@@ -1149,7 +1157,6 @@
                                          GLboolean transpose,
                                          const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, GLboolean transpose = %u, "
         "const GLfloat *value = 0x%016" PRIxPTR ")",
@@ -1158,6 +1165,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniformMatrix4fv, context, program, location, count, transpose, value);
         if (context->skipValidation() ||
             ValidateProgramUniformMatrix4fv(context, program, location, count, transpose, value))
         {
@@ -1172,7 +1180,6 @@
                                            GLboolean transpose,
                                            const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, GLboolean transpose = %u, "
         "const GLfloat *value = 0x%016" PRIxPTR ")",
@@ -1181,6 +1188,8 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniformMatrix4x2fv, context, program, location, count, transpose,
+                      value);
         if (context->skipValidation() ||
             ValidateProgramUniformMatrix4x2fv(context, program, location, count, transpose, value))
         {
@@ -1195,7 +1204,6 @@
                                            GLboolean transpose,
                                            const GLfloat *value)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint program = %u, GLint location = %d, GLsizei count = %d, GLboolean transpose = %u, "
         "const GLfloat *value = 0x%016" PRIxPTR ")",
@@ -1204,6 +1212,8 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ProgramUniformMatrix4x3fv, context, program, location, count, transpose,
+                      value);
         if (context->skipValidation() ||
             ValidateProgramUniformMatrix4x3fv(context, program, location, count, transpose, value))
         {
@@ -1214,12 +1224,12 @@
 
 void GL_APIENTRY SampleMaski(GLuint maskNumber, GLbitfield mask)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint maskNumber = %u, GLbitfield mask = 0x%X)", maskNumber, mask);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(SampleMaski, context, maskNumber, mask);
         if (context->skipValidation() || ValidateSampleMaski(context, maskNumber, mask))
         {
             context->sampleMaski(maskNumber, mask);
@@ -1234,7 +1244,6 @@
                                          GLsizei height,
                                          GLboolean fixedsamplelocations)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLenum target = 0x%X, GLsizei samples = %d, GLenum internalformat = 0x%X, GLsizei width "
         "= %d, GLsizei height = %d, GLboolean fixedsamplelocations = %u)",
@@ -1244,6 +1253,8 @@
     if (context)
     {
         TextureType targetPacked = FromGLenum<TextureType>(target);
+        ANGLE_CAPTURE(TexStorage2DMultisample, context, targetPacked, samples, internalformat,
+                      width, height, fixedsamplelocations);
         if (context->skipValidation() ||
             ValidateTexStorage2DMultisample(context, targetPacked, samples, internalformat, width,
                                             height, fixedsamplelocations))
@@ -1256,13 +1267,13 @@
 
 void GL_APIENTRY UseProgramStages(GLuint pipeline, GLbitfield stages, GLuint program)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint pipeline = %u, GLbitfield stages = 0x%X, GLuint program = %u)", pipeline, stages,
           program);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(UseProgramStages, context, pipeline, stages, program);
         if (context->skipValidation() ||
             ValidateUseProgramStages(context, pipeline, stages, program))
         {
@@ -1273,12 +1284,12 @@
 
 void GL_APIENTRY ValidateProgramPipeline(GLuint pipeline)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint pipeline = %u)", pipeline);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ValidateProgramPipeline, context, pipeline);
         if (context->skipValidation() || ValidateValidateProgramPipeline(context, pipeline))
         {
             context->validateProgramPipeline(pipeline);
@@ -1288,12 +1299,12 @@
 
 void GL_APIENTRY VertexAttribBinding(GLuint attribindex, GLuint bindingindex)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint attribindex = %u, GLuint bindingindex = %u)", attribindex, bindingindex);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(VertexAttribBinding, context, attribindex, bindingindex);
         if (context->skipValidation() ||
             ValidateVertexAttribBinding(context, attribindex, bindingindex))
         {
@@ -1308,7 +1319,6 @@
                                     GLboolean normalized,
                                     GLuint relativeoffset)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint attribindex = %u, GLint size = %d, GLenum type = 0x%X, GLboolean normalized = %u, "
         "GLuint relativeoffset = %u)",
@@ -1318,6 +1328,8 @@
     if (context)
     {
         VertexAttribType typePacked = FromGLenum<VertexAttribType>(type);
+        ANGLE_CAPTURE(VertexAttribFormat, context, attribindex, size, typePacked, normalized,
+                      relativeoffset);
         if (context->skipValidation() ||
             ValidateVertexAttribFormat(context, attribindex, size, typePacked, normalized,
                                        relativeoffset))
@@ -1332,7 +1344,6 @@
                                      GLenum type,
                                      GLuint relativeoffset)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLuint attribindex = %u, GLint size = %d, GLenum type = 0x%X, GLuint relativeoffset = "
         "%u)",
@@ -1342,6 +1353,7 @@
     if (context)
     {
         VertexAttribType typePacked = FromGLenum<VertexAttribType>(type);
+        ANGLE_CAPTURE(VertexAttribIFormat, context, attribindex, size, typePacked, relativeoffset);
         if (context->skipValidation() ||
             ValidateVertexAttribIFormat(context, attribindex, size, typePacked, relativeoffset))
         {
@@ -1352,12 +1364,12 @@
 
 void GL_APIENTRY VertexBindingDivisor(GLuint bindingindex, GLuint divisor)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLuint bindingindex = %u, GLuint divisor = %u)", bindingindex, divisor);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(VertexBindingDivisor, context, bindingindex, divisor);
         if (context->skipValidation() ||
             ValidateVertexBindingDivisor(context, bindingindex, divisor))
         {